### PR TITLE
use Mod:revert_ordering_on_local_key in {get,delete}_data for DESC keys

### PR DESCRIPTION
**Pull request by hmmr**
Depends on https://github.com/basho/riak_ql/pull/162.

Single-key primitive operations in `riak_kv_ts_api` need to provide to `riak_client:get` and `:delete` keys adjusted for fields declared DESC in the DDL, as is done for list_keys in https://github.com/basho/riak_kv/blob/riak_ts-develop/src/riak_kv_qry_worker.erl#L190.

This fixes regressions currently seen in `ts_simple_single_key_ops` (and possibly others).

### CODE REVIEW

**Comment by davidx** [comment]
thumbot retry

**Comment by davidx** [comment]
thumbot retry

File: src/riak_kv_qry_worker.erl [hunk 1]
```diff
@@ -187,10 +187,8 @@ make_key_conversion_fun(Table) ->
     Mod = riak_ql_ddl:make_module_name(Table),
     DDL = Mod:get_ddl(),
     fun(Key) when is_binary(Key) ->
-            {ok, DescAdjustedKey} =
-                riak_ql_ddl:negate_if_desc(
-                  tuple_to_list(sext:decode(Key)), Mod, DDL),
-            {ok, PK} = riak_ql_ddl:lk_to_pk(DescAdjustedKey, Mod, DDL),
+            {ok, PK} = riak_ql_ddl:lk_to_pk(
```
Comment (macintux) [inline src/riak_kv_qry_worker.erl]: Where will the `bad_compound_key` exception be handled in this case?

Comment (hmmr) [reply src/riak_kv_qry_worker.erl]: Over there: https://github.com/basho/riak_kv/pull/1560/commits/b6f142e66433d844fc01d3e982546fa2bca1c63f#diff-47d6b445c0ee52daa420d1d48d98f394R638

Comment (macintux) [reply src/riak_kv_qry_worker.erl]: This function is passed from a `riak_kv_index_fsm` process to a `riak_kv_vnode` to be invoked in a fold. I don't see any exception handling in a quick scan of that code path, which leads me to believe that whatever provokes that exception could bring down a vnode.

Comment (hmmr) [reply src/riak_kv_qry_worker.erl]: In contrast to its use in `riak_kv_ts_api:{get,delete}_data` where the key comes from user input, the folding functions here will have the key that's guaranteed to be valid (it is read directly from the backend).

Comment (macintux) [reply src/riak_kv_qry_worker.erl]: Ok. Please add a comment here, though, so someone in the future doesn't have the same panic attack.

**Comment by hazen** [comment]
:+1:

**Comment by andytill** [comment]
@thumbot retry

**Comment by andytill** [comment]
+1

**Pull request merged**